smart-contracts/wasm-chain-integration/src/v1/trie/api.rs

```diff
@@ -203,6 +237,32 @@ impl PersistentState {
     }
 }
 
+/// Iterator over all (key, value) pairs stored in the persistent state.
+/// Values are returned in increasing order of keys.
+pub struct PersistentStateIterator<'a, L> {
+    state:    MutableTrie,
+    iterator: Option<low_level::Iterator>,
+    loader:   &'a mut L,
+}
+
+impl<'a, L: BackingStoreLoad> Iterator for PersistentStateIterator<'a, L> {
+    type Item = (Vec<u8>, Vec<u8>);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let iterator = self.iterator.as_mut()?;
+        match self.state.next(self.loader, iterator, &mut EmptyCounter) {
+            Ok(v) => {
+                let entry = v?;
+                let key = iterator.get_key();
+                self.state.with_entry(entry, self.loader, |value| (key.to_vec(), value.to_vec()))
+            }
+            Err(empty) => match empty {},
+        }
+    }
+}
+
+impl<'a, L: BackingStoreLoad> FusedIterator for PersistentStateIterator<'a, L> {}
```

Comment:
Please add a comment about why this instance is useful.

Reply:
Will do.